packages/react-storage/src/components/StorageBrowser/views/LocationActionView/CreateFolderControls.tsx

```diff
@@ -77,31 +82,38 @@ export const CreateFolderControls = (): React.JSX.Element => {
     handleCreateAction({ prefix: '', options: { reset: true } });
   };
 
-  const primaryProps =
+  const actionStartProps: ActionStartProps =
     result?.status === 'COMPLETE'
       ? {
-          onClick: () => {
-            handleClose();
-          },
-          children: 'Folder created',
+          onClick: handleClose,
+          label: 'Folder created',
         }
       : {
-          onClick: () => {
-            handleCreateFolder();
-          },
-          children: 'Create Folder',
-          disabled: !folderName || !!fieldValidationError,
+          onClick: handleCreateFolder,
+          label: 'Create Folder',
+          isDisabled: !folderName || !!fieldValidationError,
         };
 
+  // FIXME: Eventually comes from useView hook
+  const contextValue: ControlsContext = {
+    data: {},
+    actionsConfig: {
+      type: 'SINGLE_ACTION',
+      isCancelable: true,
+      actionStart: {
+        ...actionStartProps,
+      },
+    },
+  };
   return (
-    <>
+    <ControlsContextProvider {...contextValue}>
       <Exit
         onClick={() => {
           handleClose();
         }}
       />
       <Title />
-      <Primary {...primaryProps} />
+      <ActionStartControl className={`${CLASS_BASE}__primary`} />
```

Comment:
Here, where the control is actually used, we should give it somewhat unique class names to its particular "usage" within a view. For example, since it's used here in the create folder view...
```suggestion
      <ActionStartControl className={`${CLASS_BASE}__create-folder-action-start`} />
```